src/encoded/upgrade/user.py

```diff
@@ -42,3 +42,19 @@ def user_7_8(value, system):
             value['viewing_groups'].append('community')
     else:
         value['viewing_groups'] = ['community']
+
+
+@upgrade_step('user', '8', '9')
+def user_8_9(value, system):
+    if 'groups' in value:
+        new_groups = []
+        for grp in value['groups']:
+            if grp == 'admin':
+                new_groups.append(grp)
+            elif grp == 'read-only-admin':
+                new_groups.append(grp)
+            elif grp == 'verified':
+                new_groups.append(grp)
+            else:
+                pass
```

Comment:
I think this could be rewritten using list comprehension:
new_groups = [group for group in value['groups'] if group in ['admin', 'verified', 'admin-read-only']]